docs/en/resource/introduction.md

```diff
@@ -0,0 +1,101 @@
+| **Single data source**                    | Resources allows you to create a single resource for a given template and share it with multiple widgets via the data middleware.                                            |
+| **Data transforms**                       | A data tranform mechanism allows you to specify the data form that your widget requires and transforms the data transparently                                                |
+| **Support for async and sync data reads** | The resource template can read data in any way it sees fit, the data middleware uses a reactive API to invalidate widgets when data becomes available.                       |
+| **Consistent Resource Options**           | Resource options objects are passed to all api functions ensuring that all api functions are pure and provide only the data that was requested                               |
+| **Sharable Resource Options**             | Resource options can be shared between widgets via the data middleware. This allows multiple widgets to react to changes to the resource options, ie. query, page number etc |
+
+# Basic Usage
+
+Dojo resource is comprised of a resource which you create to get and read data and a data middleware which you use within your widgets to access the resource.
+
+## Creating a resource
+
+A basic resource requires only a `DataTemplate` to be created which specifies a `read` function.
+
+> main.ts
+
+```ts
+import { DataTemplate, createResource } from '@dojo/framework/core/resource';
+import { fetcher } from './personfetcher';
+
+const template: DataTemplate = {
+	read: fetcher
+};
+
+const resource = createResource(template);
+```
+
+When a call is made to the resource api which requires data to be read, the read function will be called with the readoptions, a put function and a get function. These two helper functions can be used to side load data or to read existing data from the resource. More information on this can be found in the supplemental docs.
+
+## Transforming Data
+
+Resources provides a `createTransformer` function which can be used to create a type safe transform object. In order to correctly type the transform source values, a generic must be passed to the `DataTemplate` to type the `read` return type. When used with the `createTransformer` function, this ensures that only return values from the read function can be used as source values.
+
+> main.ts
+
+```ts
+import { DataTemplate, createResource, createTransformer } from '@dojo/framework/core/resource';
+import { fetcher } from './personfetcher';
+
+const template: DataTemplate<{ firstName: string; lastName: string }> = {
+	read: fetcher
+};
+
+const transformer = createTransformer(template, {
+	value: ['firstName'],
+	label: ['firstName', 'lastName']
+});
+```
+
+The transformer can be passed to a widget which uses the data middleware to ensure that the data it receives from the resource is correct.
+
+## Passing a Resource to a Widget
```

Comment:
Should this come before showing the use of the data middleware? Seems like this would flow better being after the next section?